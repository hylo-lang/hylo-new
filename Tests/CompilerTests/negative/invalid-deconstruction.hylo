//! no-std

struct A {}

trait P {
  fun f() -> {Void, ...A}
}

public fun test<T is P>(_ x: T) {
  let (a, b) = x.f()
}
