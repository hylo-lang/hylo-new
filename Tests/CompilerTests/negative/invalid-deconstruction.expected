negative/invalid-deconstruction.hylo:6.14-26: warning: open-ended tuple '{Void, ...A}' is uninhabited
  fun f() -> {Void, ...A}
             ~~~~~~~~~~~~
negative/invalid-deconstruction.hylo:10.7-13: error: tuple pattern cannot match open-ended value of type '{Void, ...A}'
  let (a, b) = x.f()
      ~~~~~~
